lib/vcard/dirinfo.rb

```diff
@@ -31,16 +31,27 @@ class DirectoryInfo
     # Initialize a DirectoryInfo object from +fields+. If +profile+ is
     # specified, check the BEGIN/END fields.
     def initialize(fields, profile = nil) #:nodoc:
-      if fields.detect { |f| ! f.kind_of? DirectoryInfo::Field }
-        raise ArgumentError, "fields must be an array of DirectoryInfo::Field objects"
+      @valid = true
+      @fields = []
+
+      fields.each do |f|
+        raise ArgumentError, "fields must be an array of DirectoryInfo::Field objects" unless f.kind_of? DirectoryInfo::Field
```

Comment:
I didn't change behavior in this place. It looks for me like a strict type check and it's not clear for me why it needed here.